test/lint/lint-python-dead-code.sh

```diff
@@ -0,0 +1,23 @@
+#!/usr/bin/env bash
+#
+# Copyright (c) 2021 The Bitcoin Core developers
+# Distributed under the MIT software license, see the accompanying
+# file COPYING or http://www.opensource.org/licenses/mit-license.php.
+#
+# Find dead Python code.
+
+export LC_ALL=C
+
+if ! command -v vulture > /dev/null; then
+    echo "Skipping Python dead code linting since vulture is not installed. Install by running \"pip3 install vulture\""
+    exit 0
+fi
+
+# --min-confidence 100 will only report code that is guaranteed to be unused within the analyzed files.
+# Any value below 100 introduces the risk of false positives, which would create an unacceptable maintenance burden.
+if ! vulture \
+    --min-confidence 100 \
+    $(git rev-parse --show-toplevel); then
```

Comment:
Should this be `$(vulture --min-confidence 100 -- $(git ls-files -- "*.py"))` instead to make sure only files in the repo are checked?

Reply:
Could be changed if people think that's better. The `rev-parse` version is what the previous incarnation of the linter had. I was thinking maybe you want to run the linter against a new file before you `git add` it? 

AFAICT, vulture automatically filters by file extension (https://github.com/jendrikseipp/vulture/blob/master/vulture/config.py#L105), so `*.py` should be implicit

Reply:
don't think this is a serious issue but, ostensibly a developer could, unwisely, keep their own `.py` scripts inside the repo directory, which would make no sense to check here, so i think `ls-files` to only check the committed ones makes sense

Reply:
Alright, changed to `ls-files` as requested